### ISSUE TITLE
Windows 10 x86 - Building Rizin with clang and llvm-rc linker is not possible

### ISSUE THREAD

**Issue by caribpa** [rizinorg/rizin#730]
### Work environment

| Questions                                            | Answers
|------------------------------------------------------|--------------------
| OS/arch/bits (mandatory)                             | Windows 10 x86
| File format of the file you reverse (mandatory)      | n/a
| Architecture/bits of the file (mandatory)            | n/a
| `rizin -v` full output, **not truncated** (mandatory)  | n/a

### Expected behavior
Building Rizin with the `clang` compiler and `llvm-rc` linker works.

### Actual behavior
Building Rizin with the `clang` compiler and `llvm-rc` linker doesn't work. The errors are:
```
[...]
C:\Program Files\Windows Kits\10\include\10.0.19041.0\um\winnt.h:6630:1: error: conflicting types for '_bittest'
_bittest (
^
C:\Program Files\Windows Kits\10\include\10.0.19041.0\um\winnt.h:6630:1: note: '_bittest' is a builtin with type 'unsigned char (const long *, long)'
C:\Program Files\Windows Kits\10\include\10.0.19041.0\um\winnt.h:7459:1: error: conflicting types for '_mm_prefetch'
_mm_prefetch (
^
C:\Program Files\Windows Kits\10\include\10.0.19041.0\um\winnt.h:7459:1: note: '_mm_prefetch' is a builtin with type 'void (const char *, int)'
[...]
C:\Program Files\Windows Kits\10\include\10.0.19041.0\ucrt\corecrt_wstdlib.h:375:27: error: conflicting types for '_wperror'
    _ACRTIMP void __cdecl _wperror(
                          ^
C:\Program Files\Windows Kits\10\include\10.0.19041.0\ucrt\corecrt_wstdio.h:168:27: note: previous declaration is here
    _ACRTIMP void __cdecl _wperror(
                          ^
[...]
C:\Program Files\Windows Kits\10\include\10.0.19041.0\ucrt\stdlib.h:236:27: error: conflicting types for 'perror'
    _ACRTIMP void __cdecl perror(_In_opt_z_ char const* _ErrMsg);
                          ^
C:\Program Files\Windows Kits\10\include\10.0.19041.0\ucrt\stdio.h:325:27: note: previous declaration is here
    _ACRTIMP void __cdecl perror(
                          ^
[...]
```

### Steps to reproduce the behavior
```cmd
> git clone https://github.com/rizinorg/rizin
> cd rizin
> set CC=clang
> set CCX=clang++
> set WINDRES=llvm-rc
# Modify C:\Users\%USER%\AppData\Local\Programs\Python\Python38-32\Lib\site-packages\mesonbuild\modules\windows.py adding the line from https://github.com/mesonbuild/meson/pull/8327/commits/fb6b706782e4651d845f1db406ca47c95bfa8161 so that llvm-rc is recognized, and remove the windows.cpython file from the __pycache__ folder
# Execute `find . -type f -exec sed -i 's/ssize_t/rz_&/g' {} +` from a msys2 shell as showed in #726

> meson --default-library=static --buildtype=release build
> ninja -C build
# 💔 - Conflicting types error and way too many warnings
```

### Additional Logs, screenshots, source code,  configuration dump, ...
Attached is the full log: [ninja.log](https://github.com/rizinorg/rizin/files/6049356/ninja.log)

Note that this is not a Rizin specific issue, but clang one, though I am filing this issue here as it affects Rizin.

Also, I'm in the process of filing a bug to https://bugs.llvm.org as this issue is also present in [Clang 11](https://releases.llvm.org/download.html#11.0.0) (the latest one with a Windows x86 installer), and the same issue happens in the Clang obtained from the VS 2019 optional package installer (Clang 10).

**Comment by XVilka** [comment]
> Also, I'm in the process of filing a bug to https://bugs.llvm.org 

@caribpa have you had any luck with that? Please write here the link to the bug then.

**Comment by caribpa** [comment]
Hey @XVilka, unfortunately I never had the account confirmed (and I completely forgot after a week of waiting) so I could never file that bug. Sorry 😿

**Comment by XVilka** [comment]
I have an account, you could write the report here, I will copypaste.

**Comment by caribpa** [comment]
I know I have a file somewhere with a little description about this issue as I wrote it while waiting for the account to be confirmed, but I cannot find it! (Maybe I have a backup, I'll try again later).

To make it even less fun, I no longer have the Windows VM I used to make this test (as a few months ago I went [Marie Kondo](https://en.wikipedia.org/wiki/Marie_Kondo) and formatted my machine), which means that I cannot re-create the *ninja.log* with the latest Clang (v12).

The best I could do in the worst case scenario (can't find the file in my backups) is rewriting the report using the *ninja.log* attached above (which is possibly really outdated), but sadly it won't happen soon (not this week).

Tell me if that's something you are happy with. 🗻
